File: bindings/lua/Polycode/AudioStreamingSource.lua
class "AudioStreamingSource"







function AudioStreamingSource:AudioStreamingSource(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycore.AudioStreamingSource(unpack(arg))
	end
end

function AudioStreamingSource:getNumChannels()
	local retVal =  Polycore.AudioStreamingSource_getNumChannels(self.__ptr)
	return retVal
end

function AudioStreamingSource:getFrequency()
	local retVal =  Polycore.AudioStreamingSource_getFrequency(self.__ptr)
	return retVal
end

function AudioStreamingSource:__delete()
	if self then Polycore.delete_AudioStreamingSource(self.__ptr) end
end

File: bindings/javascript/Polycode/AudioStreamingSource.js
function AudioStreamingSource(channels,freq) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.AudioStreamingSource(channels,freq)
	}
}


Duktape.fin(AudioStreamingSource.prototype, function (x) {
	if (x === AudioStreamingSource.prototype) {
		return;
	}
	Polycode.AudioStreamingSource__delete(x.__ptr)
})

AudioStreamingSource.prototype.getNumChannels = function() {
	return Polycode.AudioStreamingSource_getNumChannels(this.__ptr)
}

AudioStreamingSource.prototype.getFrequency = function() {
	return Polycode.AudioStreamingSource_getFrequency(this.__ptr)
}
